src/script_lua.c

```diff
@@ -656,6 +659,10 @@ static void luaReplyToRedisReply(client *c, client* script_client, lua_State *lu
 static int luaRedisGenericCommand(lua_State *lua, int raise_error) {
     int j, argc = lua_gettop(lua);
     scriptRunCtx* rctx = luaGetFromRegistry(lua, REGISTRY_RUN_CTX_NAME);
+    if (!rctx) {
+        luaPushError(lua, "redis.call/pcall can only be called inside a function invocation");
+        return raise_error ? luaRaiseError(lua) : 1;
+    }
```

Comment:
this code is not specific to functions, right?
so maybe in EVAL this `if` is not reachable, but it still looks a bit odd to refer to "functions" here.

Reply:
Fixed